src/components/Axis/Axis.jsx

```diff
@@ -0,0 +1,176 @@
+import _ from 'lodash';
+import React from 'react';
+import { lucidClassNames } from '../../util/style-helpers';
+import { discreteTicks } from '../../util/chart-helpers';
+import { createClass } from '../../util/component-types';
+
+const cx = lucidClassNames.bind('&-Axis');
+
+const {
+	array,
+	func,
+	number,
+	oneOf,
+	any,
+} = React.PropTypes;
+
+/**
+ * {"categories": ["visualizations", "chart primitives"]}
+ *
+ * Axes and allies
+ *
+ * This component is a very close sister to `d3.avg.axis` and most of the logic
+ * was ported from d3.
+ */
+const Axis = createClass({
+	displayName: 'Axis',
+
+	statics: {
+		_lucidIsPrivate: true,
+	},
+
+	propTypes: {
+		/**
+		 * Classes are appended to root element along with existing classes using
+		 * the `classnames` library.
+		 */
+		className: any,
+		/**
+		 * Must be a D3 scale.
+		 */
+		scale: func.isRequired,
+		/**
+		 * Size of the ticks for each discrete tick mark.
+		 */
+		innerTickSize: number,
+		/**
+		 * Size of the tick marks found at the beginning and end of the axis. It's
+		 * common to set this to `0` to remove them.
+		 */
+		outerTickSize: number,
+		/**
+		 * An optional function that can format ticks. Generally this shouldn't be
+		 * needed since d3 has very good default formatters for most data.
+		 *
+		 * Signature: `(tick) => {}`
+		 */
+		tickFormat: func,
+		/**
+		 * If you need fine grained control over the axis ticks, you can pass them
+		 * in this array.
+		 */
+		ticks: array,
+		/**
+		 * Determines the spacing between each tick and its text.
+		 */
+		tickPadding: number,
+		/**
+		 * Determines the orientation of the ticks. `left` and `right` will
+		 * generate a vertical axis, whereas `top` and `bottom` will generate a
+		 * horizontal axis.
+		 */
+		orient: oneOf(['top', 'bottom', 'left', 'right']),
+		/**
+		 * Control the number of ticks displayed.
+		 *
+		 * Usually when using an ordinal scale you should show a tick mark for
+		 * every value, but there are some rare cases when you want to only show a
+		 * sampling of the ticks. By passing in a number to we'll generate an
+		 * evenly spaced number of ticks and display them for ordinal and
```

Comment:
Yeah it is. When working with linear scales, this number is only a "hint" (powered by d3), but when working with ordinal scales, it's an absolute number (powered by my own util function).